frontend/src/components/RelatedObjectsSelection/RelatedObjectsSelection.stories.tsx

```diff
@@ -0,0 +1,286 @@
+import { delay } from "@maykin-ui/client-common";
+import { Meta, StoryObj } from "@storybook/react-vite";
+import { expect, userEvent, waitFor, within } from "storybook/test";
+
+import { ModalServiceDecorator } from "../../../.storybook/decorators";
+import {
+  FIXTURE_DESTRUCTION_LIST_ITEM,
+  beoordelaarFactory,
+  destructionListFactory,
+  destructionListItemFactory,
+  recordManagerFactory,
+  relatedObjectsSelectionItemsFactory,
+} from "../../fixtures";
+import { RelatedObjectsSelectionItemMutation } from "../../lib/api/relatedObjectsSelection";
+import { ZaakObject } from "../../types";
+import { RelatedObjectsSelection as RelatedObjectsSelectionComponent } from "./RelatedObjectsSelection";
+import { RelatedObjectsSelectionModal as RelatedObjectsSelectionModalComponent } from "./RelatedObjectsSelectionModal";
+
+const meta: Meta<typeof RelatedObjectsSelectionComponent> = {
+  title: "Components/RelatedObjectsSelection",
+  component: RelatedObjectsSelectionComponent,
+  decorators: [ModalServiceDecorator],
+};
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+/**
+ * Dynamic mock objects, get updated by PATCH mock, resulting in updated GET mock.
+ */
+const INITIAL_MOCK_STATE = Object.freeze(
+  relatedObjectsSelectionItemsFactory([
+    {
+      url: "https://openzaak.test.maykin.opengem.nl/zaken/api/v1/zaakobjecten/00000000-0000-0000-0000-000000000000",
+      selected: true,
+      supported: true,
+      result: {
+        relatieomschrijving: "Zaakobject 1",
+      } as ZaakObject,
+    },
+    {
+      url: "https://openzaak.test.maykin.opengem.nl/zaken/api/v1/zaakobjecten/11111111-1111-1111-1111-111111111111",
+      selected: false,
+      supported: true,
+      result: {
+        relatieomschrijving: "Zaakobject 2",
+      } as ZaakObject,
+    },
+    {
+      url: "https://openzaak.test.maykin.opengem.nl/zaken/api/v1/zaakobjecten/22222222-2222-2222-2222-222222222222",
+      selected: false,
+      supported: false,
+      result: {
+        relatieomschrijving: "Zaakobject 3",
+      } as ZaakObject,
+    },
+  ]),
+);
+
+let MOCK_STATE = INITIAL_MOCK_STATE;
```

Comment:
Some state management to mimic the behavior of the `RelatedObjectsView` endpoint.